docs/access-control.md

```diff
@@ -0,0 +1,71 @@
+# Multi-tenant Support and Access Control
+Multi-tenant support and access control is provided by the [openshift-elasticsearch-plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin).  This plugin utilizes [SearchGuard](https://github.com/floragunncom/search-guard) to define and maintain role based access control lists.  Roles are generated based upon the Openshift projects that are visible to an Openshift user.  The following is a brief explanation how roles and role mappings are generated.  Please see the [official SearchGuard documentation](http://floragunncom.github.io/search-guard-docs) for additional information related to SearchGuard.
+
+## Role Definitions and Permissions
+The permissions for a user are generated when an authenticated user makes their first request to Elasticsearch.  This request typically originates from the Kibana user interface provided by the Openshift logging stack.  The `openshift-elasticsearch-plugin` retrieves the list of projects, on behalf of a user, that are visible to the user.  The plugin will generate roles for each user similiar to:
+
+```
+gen_ocp_kibana_shared:
+  cluster: [CLUSTER_MONITOR_KIBANA]
+  indices:
+    '*':
+      '*': [INDEX_ANY_KIBANA]
+gen_project_operations:
+  cluster: [CLUSTER_OPERATIONS]
+  indices:
+    '*?*?*':
+      '*': [INDEX_ANY_OPERATIONS]
+    ?operations?:
+      '*': [INDEX_OPERATIONS]
+gen_user_user2_bar_email_com:
+  indices:
+    .all_user2_bar_email_com:
+      '*': [INDEX_PROJECT]
+    ?kibana?994a33f6a157ba4a286395f81a4333db1e6cefb6:
+      '*': [INDEX_KIBANA]
+    ?project?foo?bar?*:
```

Comment:
these don't need to begin with `?`

Reply:
This is a straight port from refactoring of the roles.  I believe the ? also means the dot is optional and I wondered why we had it too.  We will need to make a change to the plugin to correct.  I'll update the documentation to remove it for now